### ISSUE TITLE
Project Admin - Validate all tasks messes up "Contributions" on Activity and Stats page

### ISSUE THREAD

**Issue by bgirardot** [hotosm/tasking-manager#986]
If you use the Validate all tasks button in the project admin, it seems to make the system think the person who clicked "Validate all tasks" is the person who mapped and validated every task as shown in the Activity and Stats page.

It makes sense that 'validate all tasks' should have some affect on the contributions field on activity and stats, but it needs to work differently.

![2017-12-09_14h01_58](https://user-images.githubusercontent.com/265831/33795891-1ad6059a-dcea-11e7-8bf2-5c01b1bd6bc4.png)

**Comment by sunidhiraheja** [comment]
Working on this!

**Comment by pgadige** [comment]
I observed that the `validate_all_tasks` method extracts all tasks belonging to a specific project and selectively filters those having statuses `READY`, `LOCKED_FOR_MAPPING`,`MAPPED`, `LOCKED_FOR_VALIDATING`,`VALIDATED`

regardless of what status the task is in, the method explicitly sets the project admin as the task's mapper, `task.mapped_by`. I think this might be causing the mapped column in contributions table to show all tasks being mapped by the project admin.

If the task is locked either for mapping or validation purposes, the method unlocks the task, sets it's new status to `VALIDATED` and, eventually, makes the project admin as the task's validator, `task.validated_by`

If the task is in any of these statuses - `READY`, `MAPPED` or `VALIDATED`, it is locked for validation. I wonder why a task in `READY` is being locked for validation when it is yet to be mapped. I think we have to selectively eliminate all `READY` tasks just the way we eliminated `BADIMAGERY` ones?
Subsequently, the task is unlocked after pushing it to `VALIDATED` status and making the project admin it's validator.

I'd like to work on this task.

**Comment by sunidhiraheja** [comment]
@pgadige Hi, I'm working on this issue.
Thanks for pointing out the possible problem involving validation of all tasks except those with status BAD_IMAGERY.
Do you think you should create a new issue about this?
I feel like this decision of task types needs discussion. For instance, do we want invalidated tasks to be remapped before they can be validated?

**Comment by bgirardot** [comment]
@sunidhiraheja When we click on the "Validate all Tasks" button, we want all tasks to get marked validated, regardless of their state.

This is for use when we have gone outside of the tasking manager to make sure the area has been mapped and the mapping is validated mapping. This is easy to do in the JOSM editor so Project Managers do it quite often instead of doing it through the Tasking Manager.

The only question is how do we handle "Bad Imagery" tasks, but at the moment, this issue is to fix the problem we have that we lose the information about who mapped individual tasks and who validated individual tasks. That info can not get lost. When a Project Manager clicks "Validate all Tasks" it should not be changing any of the existing info like who marked it mapped or who marked it validated. 

It does not matter very much if all the tasks that were not validated and/or not mapped get info about the Project Manager who is clicking "Validate all Tasks", it is fine if all the tasks that were not already mapped and/or validated get that project managers name, but it is not super important.

**Comment by sunidhiraheja** [comment]
Sorry, I accidentally deleted my last comment.

So is it okay if I set the `mapped_by` value to the pm's ID if its not already set?

**Comment by bgirardot** [comment]
@sunidhiraheja Yes, you can set it to the PM's id if it is not already set, both for mapped and for validated.

(sorry I am not much of a programmer, so am not much help code or programming related, just usage and use cases and testing :)
